app/feature/feature-movingflow/src/main/kotlin/com/hedvig/android/feature/movingflow/MovingFlowGraph.kt

```diff
@@ -0,0 +1,146 @@
+
+  @Serializable
+  data object CompareCoverage : MovingFlowDestinations, Destination
+
+  @Serializable
+  data class Summary(
+    val moveIntentId: String,
+    val homeQuoteId: String,
+  ) : MovingFlowDestinations, Destination
+
+  @Serializable
+  data class SuccessfulMove(
+    val moveDate: LocalDate,
+  ) : MovingFlowDestinations, Destination {
+    companion object : DestinationNavTypeAware {
+      override val typeList: List<KType> = listOf(typeOf<LocalDate>())
+    }
+  }
+}
+
+fun NavGraphBuilder.movingFlowGraph(navController: NavController, onNavigateToNewConversation: () -> Unit) {
+  navgraph<MovingFlowGraphDestination>(
+    startDestination = MovingFlowDestinations.Start::class,
+  ) {
+    navdestination<MovingFlowDestinations.Start> {
+      StartDestination(
+        viewModel = koinViewModel<StartViewModel>(),
+        navigateUp = navController::navigateUp,
+        onNavigateToNextStep = { moveIntentId ->
+          navController.navigate(EnterNewAddress(moveIntentId))
+        },
+      )
+    }
+    navdestination<MovingFlowDestinations.EnterNewAddress> {
+      val moveIntentId = it.toRoute<MovingFlowDestinations.EnterNewAddress>().moveIntentId
+      EnterNewAddressDestination(
+        viewModel = koinViewModel<EnterNewAddressViewModel>(),
+        navigateUp = navController::navigateUp,
+        popBackStack = navController::popBackStack,
+        onNavigateToAddHouseInformation = {
+          navController.navigate(MovingFlowDestinations.AddHouseInformation(moveIntentId))
+        },
+        onNavigateToChoseCoverageLevelAndDeductible = {
+          navController.navigate(MovingFlowDestinations.ChoseCoverageLevelAndDeductible(moveIntentId))
+        },
+      )
+    }
+    navdestination<MovingFlowDestinations.AddHouseInformation> {
+      AddHouseInformationDestination(
+        viewModel = koinViewModel<AddHouseInformationViewModel>(),
+        navigateUp = navController::navigateUp,
+        popBackStack = navController::popBackStack,
+        onNavigateToChoseCoverageLevelAndDeductible = {
+          navController.navigate(MovingFlowDestinations.ChoseCoverageLevelAndDeductible(moveIntentId))
+        },
+      )
+    }
+    navdestination<MovingFlowDestinations.ChoseCoverageLevelAndDeductible> { backStackEntry ->
+      val moveIntentId = backStackEntry.toRoute<MovingFlowDestinations.ChoseCoverageLevelAndDeductible>().moveIntentId
+      ChoseCoverageLevelAndDeductibleDestination(
+        viewModel = koinViewModel<ChoseCoverageLevelAndDeductibleViewModel>(),
+        navigateUp = navController::navigateUp,
+        onNavigateToSummaryScreen = { homeQuoteId ->
+          navController.navigate(MovingFlowDestinations.Summary(moveIntentId, homeQuoteId))
+        },
+      )
+    }
+    navdestination<MovingFlowDestinations.CompareCoverage> {
+      // todo moving flow, add shared compare coverage screen
+      CompareCoverageDestination(koinViewModel<CompareCoverageViewModel>())
```

Comment:
I suppose comparison should be in shared, where forever-ui module is? We'd pass list of termsVersion strings to there as parameters

Reply:
Yeah something like that, I kept the comment vague as it will be easier to know the details once we actually know how we should approach this